pkg/controller/update_cluster.go

```diff
@@ -308,6 +309,78 @@ func (c *Controller) addPodToPlacement(cluster *myspec.M3DBCluster, pod *corev1.
 	return nil
 }
 
+func (c *Controller) checkPodsForReplacement(
+	cluster *myspec.M3DBCluster,
+	pods []*corev1.Pod,
+	pl placement.Placement) (string, *corev1.Pod, error) {
+
+	var instancePodID myspec.PodIdentity
+
+	insts := pl.Instances()
+	sort.Sort(placement.ByIDAscending(insts))
+
+	sortedPods, err := sortPods(pods)
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot sort pods: %v", err)
+	}
+
+	for _, pod := range sortedPods {
+
+		clusterPodID, err := c.podIDProvider.Identity(pod.pod, cluster)
+		if err != nil {
+			return "", nil, err
+		}
+
+		for _, inst := range insts {
+			if strings.EqualFold(strings.Split(inst.Hostname(), ".")[0], pod.pod.Name) {
```

Comment:
I like that this is more efficient and that we can skip unmarshalling unnecessarily based on this check! I wonder if it's worth skipping this check and just using the unmarshalled check below in case we ever want to use IPs for hostnames or something

Reply:
I don't think we should block merging on this since we know we only use hostname, but can we create an issue to follow up?

Reply:
yeah definitely 👍